ykcapi/src/lib.rs

```diff
@@ -61,4 +66,52 @@ mod c_testing {
         Box::leak(bm);
         ret
     }
+
+    #[no_mangle]
+    pub extern "C" fn __yktrace_start_tracing(kind: usize) -> *mut ThreadTracer {
```

Comment:
We presumably need to document somewhere that this returns a `Box`ed thing? Even if it's just a comment that says "if you don't follow this up with `__yktrace_stop_tracing` you're in undefined behaviour"?

Reply:
We can do yeah, but I've not been too worried about overly documenting code in the testing interface.

Reply:
Hmm, good point. OK I'm less worried if it's purely used for testing.